cpp/include/cudf/detail/stream_compaction.hpp

```diff
@@ -136,101 +136,20 @@ std::unique_ptr<experimental::table>
  * If `ignor_nulls` is true and `nan_as_null` is false, only `null` is
  * ignored, `NaN` is considered in unique count.
  *
- * @param[in] input         The column_view whose unique elements will be counted.
- * @param[in] ignore_nulls  flag to ignore `null` in unique count if true
- * @param[in] nan_as_null   flag to consider `NaN==null` if true.
+ * @param[in] input The column_view whose unique elements will be counted.
+ * @param[in] _include_nulls flag to include or ignore `null` while counting
+ * @param[in] nan_as_null flag to consider `NaN==null` if true.
  * @param[in] mr Optional, The resource to use for all allocations
  * @param[in] stream Optional CUDA stream on which to execute kernels
  *
  * @return number of unique elements
  */
 
 cudf::size_type unique_count(column_view const& input,
-                             bool const& ignore_nulls,
-                             bool const& nan_as_null,
+                             include_nulls const _include_nulls,
```

Comment:
@codereport suggested to use `enum class include_nulls : bool` for readability for another API. (especially for readability at calling code and unit tests). 

Also @trevorsm7 suggested to rename to `include_nulls_t`. would that be suitable here?`

I replaced `ignore_nulls` because already a replacement enum class existed. 
`nan_as_null` was used in this API only. I should create an enum class for this too.

Reply:
Or I could translate `include_nulls` to `bool` in `cudf::reduce()` code. 
I am not sure which is a better choice.